WcaOnRails/app/models/competition.rb

```diff
@@ -1,3 +1,189 @@
 class Competition < ActiveRecord::Base
   self.table_name = "Competitions"
+
+  has_many :registrations, foreign_key: "competitionId"
+  has_many :results, foreign_key: "competitionId"
+  has_many :competition_delegates
+  has_many :delegates, through: :competition_delegates
+  has_many :competition_organizers
+  has_many :organizers, through: :competition_organizers
+
+  ends_with_year_re = /\A.* \d{4}\z/
+  # TODO - learn ruby...
+  @@pattern_link_re = /\[\{([^}]+)}\{((https?:|mailto:)[^}]+)}\]/
```

Comment:
I'd use a constant for this: `PATTERN_LINK_RE = /.../` and for `pattern_text_with_links_re` as well.

Using class variables `@@` is generally frowned upon since they're prone to race conditions (global mutable state).